edx_proctoring/views.py

```diff
@@ -344,25 +348,36 @@ def get(self, request):
                         data={'detail': _('Must be a Staff User to Perform this request.')}
                     )
 
-        # If there are multiple onboarding exams, use the first exam
-        onboarding_exam = (ProctoredExam.get_practice_proctored_exams_for_course(course_id)
-                           .order_by('-created').first())
-        if not onboarding_exam or not get_backend_provider(name=onboarding_exam.backend).supports_onboarding:
+        # If there are multiple onboarding exams, use the first exam accessible to the user
+        onboarding_exams = list(ProctoredExam.get_practice_proctored_exams_for_course(course_id).order_by('-created'))
+        if not onboarding_exams or not get_backend_provider(name=onboarding_exams[0].backend).supports_onboarding:
             return Response(
                 status=404,
                 data={'detail': _('There is no onboarding exam related to this course id.')}
             )
 
+        learning_sequences_service = get_runtime_service('learning_sequences')
+        course_key = CourseKey.from_string(course_id)
         user = get_user_model().objects.get(username=(username or request.user.username))
-        serialized_onboarding_exam = ProctoredExamSerializer(onboarding_exam).data
+        details = learning_sequences_service.get_user_course_outline_details(
+            course_key, user, pytz.utc.localize(datetime.max)
```

Comment:
For user-research purposes (as the maintainers of the `learning_sequences` API), why `datetime.max`? Is there a concept the API is missing?

Reply:
Yeah, this seems to be the only way to know if a sequence will become accessible to the user later in the course. I'd use the course end date here if I had it before making this call.

Reply:
Ah, got it. So it's not enough to know whether they can know it exists now, we have to know whether they will someday have access to it. Thank you!

Minor nit: now that we're running in Python 3 only, you don't have to use pytz–you can get the max datetime in UTC with just the stdlib:

```python
from datetime import datetime, timezone

datetime.max.replace(tzinfo=timezone.utc)
```

Reply:
Actually, do we have to worry that it will close before `datetime.max`? Just trying to figure out if there's another API call we're looking for that will specifically get anything that will ever become accessible between a certain range of dates...?

Reply:
Actually yes, that's something I hadn't fully considered. The due date does not prevent a section from returning as an accessible sequence from what I've seen. Is there another mechanism which would cause sections not to be returned for `datetime.max`?